gears/src/types/tx/mod.rs

```diff
@@ -26,6 +26,35 @@ pub trait TxMessage:
     fn type_url(&self) -> &'static str;
 }
 
+/// Utility type that guarantees correctness of transaction messages set
+pub struct Messages<T: TxMessage>(pub(crate) Vec<T>);
```

Comment:
Do we really want this public across the entire crate?

Reply:
Used a function to convert the instance.